compass/landice/tests/ismip6_run/ismip6_ais_proj2300/set_up_experiment.py

```diff
@@ -268,5 +310,95 @@ def run(self):
         """
         Run this step of the test case
         """
+
+        config = self.config
+        logger = self.logger
+        section = config['ismip6_run_ais_2300']
+        sea_level_model = section.getboolean('sea_level_model')
+        if sea_level_model:
+            self._build_mapping_files(config, logger)
+
         run_model(step=self, namelist='namelist.landice',
                   streams='streams.landice')
+
+    def _build_mapping_files(self, config, logger):
+        """
+        Build mapping files between the MALI mesh and the SLM grid.
+
+        Parameters
+        ----------
+        config : compass.config.CompassConfigParser
+            Configuration options for a ismip6 forcing test case
+
+        logger : logging.Logger
+            A logger for output from the step
+        """
+
+        config = self.config
+        section = config['ismip6_run_ais_2300']
+        init_cond_path = section.get('init_cond_path')
+        nglv = section.getint('nglv')
+
+        # create the mapping files
+        # first create scrip files
+        mali_scripfile = 'mali_scripfile.nc'
+        slm_scripfile = f'slm_nglv{nglv}scripfile.nc'
+        mali_meshfile = 'mali_meshfile.nc'
+
+        # slm scripfile
+        logger.info(f'creating scripfile for the SLM grid with '
+                    f'{nglv} Gauss-Legendre points in latitude')
+
+        args = ['ncremap',
+                '-g', slm_scripfile,
+                '-G',
+                f'latlon={nglv},{2*int(nglv)}#lat_typ=gss#lat_drc=n2s']
+
+        check_call(args, logger=logger)
+
+        # mali scripfile
+        # first have to adjust lat-lon values as if they are on sphere
+        shutil.copy(init_cond_path, mali_meshfile)
+        args = ['set_lat_lon_fields_in_planar_grid.py',
+                '--file', mali_meshfile,
+                '--proj', 'ais-bedmap2-sphere']
```

Comment:
If we need to use a projection with a sphere, we should open a PR to MPAS-Tools that adds this option.  I feel like we had considered a solution that would allow the MALI mesh to keep its standard projection but couple to SLM on the sphere, but I'm having trouble remembering the details.  @hollyhan or @xylar , do you remember?

One concern I have here is that this change could complicate comparing MALI runs with and without SLM coupling.  I guess the only difference is the mapping and the MALI input files are unaffected by assuming the sphere, but it would be good to remember the details of this sphere/ellipsoid discrepancy and document that in the user's guide for the test.

Reply:
I'm not remembering the details either, even whether we talked about in detail or not.. @xylar - do you have any suggestions?

Reply:
Sorry, I clearly lost track of these comments from January.

I'm fine with @hollyhan making an MPAS-Tools PR for these changes.  I agree that the use of the lat/lon on the sphere should only affect the coupling with the SLM so it shouldn't affect comparison with a MALI run without the SLM.

> but it would be good to remember the details of this sphere/ellipsoid discrepancy and document that in the user's guide for the test.

> I'm not remembering the details either, even whether we talked about in detail or not.

Yes, we talked about the sphere vs. ellipsoid issue extensively during my fall visit to LANL.  We found that we needed  to account for it to get the expected conservative mapping between MALI and the SLM, and I think that's what @matthewhoffman wants documented.  @hollyhan, hopefully you have notes from that time that can be turned into documentation, but we can revisit those discussions if necessary.